targets/f7/furi_hal/furi_hal_usb_ccid.c

```diff
@@ -186,6 +199,12 @@ static bool connected = false;
 static bool smartcard_inserted = true;
 static CcidCallbacks* callbacks[CCID_TOTAL_SLOTS] = {NULL};
 static void* cb_ctx[CCID_TOTAL_SLOTS];
+static FuriThread* ccidThread;
```

Comment:
I think it's time to cleanup this part:
- encapsulate everything into FuriHalUsbCcid object
- allocate buffers only when you need them
- naming convention

Reply:
Will do - my main concern with buffer is that we would end up with a fragmented heap memory if we perform too many sequential allocations/de-allocations (specially given the fact we are dealing with an embedded device).

Reply:
Check furi_hal_subghz_start_async_tx, it also allocates buffer but only when subghz susbsystem is used.

Reply:
That aligns with what I had in mind - thanks